### ISSUE TITLE
order of coordinates in posList 

### ISSUE THREAD

**Issue by cnavacch** [geopython/pycsw#621]
I have tried to set up a spatial query for a CSW server, once with a `gml:Point` (````<gml:pos srsDimension='2'>48.1537 16.2825 </gml:pos>````) and once with a `gml:Polygon` (e.g., ```<gml:posList srsDimension='2'>48.1537 16.2825 48.1537 16.3759  48.2201 16.3759 48.2201 16.2825 48.1537 16.2825</gml:posList>```) geometry. I have assigned the same spatial reference system to both (EPSG:4326), which prescribes the order of the points to be latitude first, longitude second. However, the returned results differ completely in space, where the point query was right, but the polygon query was wrong. 

Having a look at the code, which parses GML in pyCSW it turns out that `gml:Point/gml:pos` are differently parsed than `gml:Polygons/gml:posList`. The following lines seem wrong to me:
https://github.com/geopython/pycsw/blob/b5a657e1e0bc7fbfb327d5635d4bdfc9d15c6f5d/pycsw/ogc/gml/gml3.py#L50-L51.
Assuming `axisorder='xy'` points are parsed latitude first, longitude second, which is okay. But in the two lines above `xlist` are the longitudes and `ylist` the latitudes, which is wrong. The following should solve the problem:
```
xlist = tmp[::2]
ylist = tmp[1::2]
```
No matter what geometry parsing is modified, at least they should be consistent. I hope I did not oversee something or got this wrong, thanks!

**Comment by tomkralidis** [comment]
Thanks for the report.  Are you able to provide a test case so I can reproduce?  The below examples are some tests against your snippet above.

```python
>>> from pycsw.ogc.gml.gml3 import _poslist2wkt
>>> _poslist2wkt('48.1537 16.2825 48.1537 16.3759 48.2201 16.3759 48.2201 16.2825 48.1537 16.2825', 'xy')
'16.2825 48.1537, 16.3759 48.1537, 16.3759 48.2201, 16.2825 48.2201, 16.2825 48.1537'
>>> _poslist2wkt('48.1537 16.2825 48.1537 16.3759 48.2201 16.3759 48.2201 16.2825 48.1537 16.2825', 'yx')
'48.1537 16.2825, 48.1537 16.3759, 48.2201 16.3759, 48.2201 16.2825, 48.1537 16.2825' 
```

**Comment by cnavacch** [comment]
I hope I got you right in setting up a test case. Below is a MWE with `Python==3.7`, `pycsw==2.4.0` and `owslib==0.20.0`:

```
from owslib.etree import etree
from pycsw.core.config import StaticContext
from pycsw.ogc.fes.fes2 import _get_spatial_operator

request_point = '<csw:GetRecords xmlns:csw="http://www.opengis.net/cat/csw/2.0.2" xmlns:gml="http://www.opengis.net/gml" ' \
                'xmlns:ogc="http://www.opengis.net/ogc" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" ' \
                'xmlns:ows="http://www.opengis.net/ows" outputSchema="http://www.opengis.net/cat/csw/2.0.2" ' \
                'outputFormat="application/xml" version="2.0.2" service="CSW" resultType="results" maxRecords="1000" ' \
                'xsi:schemaLocation="http://www.opengis.net/cat/csw/2.0.2 http://schemas.opengis.net/csw/2.0.2/CSW-discovery.xsd"> ' \
                '<csw:Query typeNames="csw:Record">' \
                '<csw:ElementSetName>full</csw:ElementSetName>' \
                '<csw:Constraint version="1.1.0">' \
                '<ogc:Filter>' \
                '<ogc:And>' \
                '<ogc:Intersects>' \
                '<ogc:PropertyName>ows:BoundingBox</ogc:PropertyName>' \
                '<gml:Point srsName="http://www.opengis.net/gml/srs/epsg.xml#4326">' \
                '<gml:pos srsDimension="2">48.1537 16.2825</gml:pos></gml:Point>' \
                '</ogc:Intersects>' \
                '</ogc:And>' \
                '</ogc:Filter>' \
                '</csw:Constraint>' \
                '</csw:Query>' \
                '</csw:GetRecords>'

request_polygon = '<csw:GetRecords xmlns:csw="http://www.opengis.net/cat/csw/2.0.2" xmlns:gml="http://www.opengis.net/gml" ' \
                  'xmlns:ogc="http://www.opengis.net/ogc" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" ' \
                  'xmlns:ows="http://www.opengis.net/ows" outputSchema="http://www.opengis.net/cat/csw/2.0.2" ' \
                  'outputFormat="application/xml" version="2.0.2" service="CSW" resultType="results" maxRecords="1000" ' \
                  'xsi:schemaLocation="http://www.opengis.net/cat/csw/2.0.2 http://schemas.opengis.net/csw/2.0.2/CSW-discovery.xsd"> ' \
                  '<csw:Query typeNames="csw:Record">' \
                  '<csw:ElementSetName>full</csw:ElementSetName>' \
                  '<csw:Constraint version="1.1.0">' \
                  '<ogc:Filter>' \
                  '<ogc:And>' \
                  '<ogc:Intersects>' \
                  '<ogc:PropertyName>ows:BoundingBox</ogc:PropertyName>' \
                  '<gml:Polygon srsName="http://www.opengis.net/gml/srs/epsg.xml#4326">' \
                  '<gml:posList srsDimension="2">48.1537 16.2825</gml:posList></gml:Polygon>' \
                  '</ogc:Intersects>' \
                  '</ogc:And>' \
                  '</ogc:Filter>' \
                  '</csw:Constraint>' \
                  '</csw:Query>' \
                  '</csw:GetRecords>'

elem_point = etree.fromstring(request_point).getchildren()[0].getchildren()[1].getchildren()[0].getchildren()[0].getchildren()[0]
elem_polygon = etree.fromstring(request_polygon).getchildren()[0].getchildren()[1].getchildren()[0].getchildren()[0].getchildren()[0]
print(_get_spatial_operator('test', elem_point, 'test', StaticContext().namespaces))
print(_get_spatial_operator('test', elem_polygon, 'test', StaticContext().namespaces))
```
I tried to circumvent to initialise a server and solely tried to test the GML translation from a given URL POST request.
I am sure there is a better way how to set this up. The print commands yield the following result:
```
query_spatial(test,'POINT(16.2825 48.1537)','intersects','false')
query_spatial(test,'POLYGON((48.1537 16.2825))','intersects','false')
```
This shows that the coordinates get swapped in case for the point (which are the coordinates I want in Lon/Lat), but not for the polygon.
Please let me know if you need further information.

**Comment by tomkralidis** [comment]
> > `query_spatial(test,'POLYGON((48.1537 16.2825))','intersects','false')`

Is this a valid polygon?  We could probably add a test in the code to ensure the minimum number of tuples (2 for line, 5 for polygon)?

**Comment by cnavacch** [comment]
I did not even think about the validity of the polygon in this case, since I wanted to demonstrate what happens when specifying the same input, i.e. 2 coordinates. The polygon is not valid, because it must contain at least 4 tuples (triangle +  first point) to be valid according to GML standards, I think. I am pretty sure that PostGIS (if used) will raise an error anyway later on. But I agree, a test at this place would be suitable to check if the points follow a specific standard.

**Comment by tomkralidis** [comment]
Further testing, a valid GetRecords request would be:
```xml
<csw:GetRecords
	xmlns:csw="http://www.opengis.net/cat/csw/2.0.2"
	xmlns:gml="http://www.opengis.net/gml"
	xmlns:ogc="http://www.opengis.net/ogc"
	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
	xmlns:ows="http://www.opengis.net/ows" outputSchema="http://www.opengis.net/cat/csw/2.0.2" 
                  outputFormat="application/xml" version="2.0.2" service="CSW" resultType="results" maxRecords="1000" 
                  xsi:schemaLocation="http://www.opengis.net/cat/csw/2.0.2 http://schemas.opengis.net/csw/2.0.2/CSW-discovery.xsd">
	<csw:Query typeNames="csw:Record">
		<csw:ElementSetName>full</csw:ElementSetName>
		<csw:Constraint version="1.1.0">
			<ogc:Filter>
				<ogc:Intersects>
					<ogc:PropertyName>ows:BoundingBox</ogc:PropertyName>
					<gml:Polygon srsName="http://www.opengis.net/gml/srs/epsg.xml#4326">
						<gml:exterior>
							<gml:LinearRing>
								<gml:posList srsDimension="2">48.1537 16.2825</gml:posList>
							</gml:LinearRing>
						</gml:exterior>
					</gml:Polygon>
				</ogc:Intersects>
			</ogc:Filter>
		</csw:Constraint>
	</csw:Query>
</csw:GetRecords>
```

which returns accordingly.  I'll add a check as discussed.

**Comment by cnavacch** [comment]
@tomkralidis : Many thanks!

**Comment by tomkralidis** [comment]
Fixed in master and 2.4 branch.
